### ISSUE TITLE
Global ZNC Timezone setting.

### ISSUE THREAD

**Issue by Red-M** [znc/znc#1202]
There is no setting to allow a default to be set by admins or for the admin/user logs.

**Comment by DarthGandalf** [comment]
There is a setting:
```
export TZ=Europe/Dublin
```
in shell before starting ZNC.

**Comment by Mikaela** [comment]
Or `TZ=Europe/Helsinki znc`, but I am in favour of global option for it (defaulting to `$TZ`).

At IRC I commented that the timezone option is deprecated because of [server-time], but I didn't think about global [log] module.

I think that it's more clear for the global [log] module to be in the timezone of ZNC admin than UTC for random users. *(UTC comes from the fact that server-time uses it, doesn't it?)*

[server-time]:http://ircv3.net/specs/extensions/server-time-3.2.html
[log]:http://wiki.znc.in/log

**Comment by DarthGandalf** [comment]
> (UTC comes from the fact that server-time uses it, doesn't it?)

No, it doesn't.

**Comment by Mikaela** [comment]
>> (UTC comes from the fact that server-time uses it, doesn't it?)


> No, it doesn't.

Am I misunderstanding [the spec](http://ircv3.net/specs/extensions/server-time-3.2.html) then?

> When enabled, the server-time extension enables optional time message tag which can be used in messages from server to client. The value of the tag MUST have the following syntax:

> `<value> ::= YYYY-MM-DDThh:mm:ss.sssZ`

> It represents a calendar date and time of day in UTC using extended format, as specified by ISO 8601:2004(E) 4.3.2. Precision of the time is one millisecond.

**Comment by DarthGandalf** [comment]
`server-time` is property of client connection, not of ZNC user.

If client supports server-time, the time is sent in UTC as per spec, and user setting is ignored.
If client doesn't support server-time, the time is sent in the configured timezone, inside text part of messages.

Log module uses the user setting too. For global log module it's not so useful.

**Comment by Mikaela** [comment]
> server-time is property of client connection, not of ZNC user.

But ZNC is a client which should also support `server-time`?

> If client supports server-time, the time is sent in UTC as per spec, and user setting is ignored.
If client doesn't support server-time, the time is sent in the configured timezone, inside text part of messages.

But ZNC connects to server while you are only talking about client and ZNC?

> Log module uses the user setting too. For global log module it's not so useful.

It would still be nice to be able to define timezone for the global log module, as I said earlier: *Or `TZ=Europe/Helsinki znc`, but I am in favour of global option for it (defaulting to `$TZ`).*

**Comment by Mikaela** [comment]
> DarthGandalf changed the title from* Global ZNC Timezone setting*. to *Global log module: timezone setting*. an hour ago

I missed this part. Is there any other global module or some reason third party module would need to be able to have global timezone or it's own option for setting timezone? If not, then I am fine with the current title.

**Comment by DarthGandalf** [comment]
> But ZNC connects to server while you are only talking about client and ZNC?

Log module ignores existence of `server-time`, even if message contains the tag, and uses local time + user's TZ setting anyway.
I'll open a separate issue about it.

**Comment by hashworks** [comment]
Do I understand this correctly, one can use `export TZ=UTC` to log in UTC?

**Comment by Mikaela** [comment]
Yes, if you then start ZNC manually. It may be better to specify in your init or cron or however you generally start ZNC.
